### DISCUSSION TITLE
How to load installed examples into tabs in Qt Design Studio?

### DISCUSSION THREAD

**Discussion by dewcked** [miurahr/aqtinstall#489] [Q&A] [answered]
[I Installed Qt and set environment for Qt Design Studio](https://gist.github.com/dewcked/fb0d4661c7298e9c6c382c87506e957e). Everything Except Qt is auto-detected in Qt Design Studio. There was no problem building project.

(edit) And I installed examples with command `aqt install-example windows desktop 6.3.0 --outputdir C:\Qt`, docs with `aqt install-doc windows 6.3.0 --outputdir C:\Qt`.

One problem is, there was no Qt examples at `Welcome > Examples` tab. I could see nothing from there. After some investigation, I found [this article](https://dev.getsol.us/T2831) explaining about dependency issue. ([Official documentation](https://doc.qt.io/qt-5/26-qdoc-configuration-example-manifest-files.html))

> we need qt5-tools for the manifest files to be created. See Qt Documentation.
> 
> the -docs packages would need to depend on qt5-tools. Since we build them with the base packages, qt5-tools depends on qt5-base and qt5-base would depend on qt5-tools.
> 
> > examples-manifest.xml:
> > 
> > These files are generated by qdoc. For example, in Qt each module has a .qdocconf which pulls in .qdoc files for the examples, which contain the documentation for the example and the meta-information for the examples-manifest file for that module.
> > 
> > the .qdocconf contains a examplesinstallpath variable, which is used as a prefix for all example paths
> > the example's .qdoc contains a tag with the path to the example
> > These two are concatenated to the relative path that ends up in the projectPath tag in the manifest file.
> > 
> > For example, in Qt src/widgets/doc/qtwidgets.qdocconf there is "examplesinstallpath = widgets", and in examples/widgets/doc/src/wiggly.qdoc there is "widgets/wiggly", which results in "widgets/widgets/wiggly/wiggly.pro" as the projectPath (which is later resolved relative to QT_INSTALL_EXAMPLES).
> 
> This could be the reason why no manifest files are created at all... or very likely will be. Will be testing more tomorrow, but my best guess is: we'll need another package structure for this features to get installed properly.
> 
> arch depends: qt5-doc depends on qt5-tools and qt5-examples on qt5-doc, which would imply the manifests are created in qt5-doc, which qt5-examples use. That's probably why there is a huge qt5-doc package with all the docs, since that could get hard to split up.

Can I automatically generate this manifest file easily or should I write a code for generating it myself?

**Comment by miurahr** [comment, marked as answer]
https://github.com/miurahr/aqtinstall/issues/448 This issue may help you.

**Reply by dewcked** [reply]
So.. it is basically the same problem. Thank you.
